mysql-test/suite/rpl/t/bug93917.test

```diff
@@ -0,0 +1,73 @@
+--echo #
+--echo # Bug #93917 "Wrong binlog entry for BLOB on a blackhole intermediary master"
+--echo # (https://bugs.mysql.com/bug.php?id=93917)
+--echo #
+
+--source include/have_innodb.inc
+--source include/have_blackhole.inc
+--source include/have_binlog_format_row.inc
+
+--source include/master-slave.inc
+
+--let $safe_load_data_dir = `SELECT @@global.secure_file_priv`
+--let $server_1_binlog_record_file = $safe_load_data_dir/bug93917_server_1.txt
+--let $server_2_binlog_record_file = $safe_load_data_dir/bug93917_server_2.txt
+
+--let $data_type_list = 'TINYBLOB', 'BLOB', 'MEDIUMBLOB', 'LONGBLOB', 'TINYTEXT', 'TEXT', 'MEDIUMTEXT', 'LONGTEXT', 'VARCHAR(64)', 'VARBINARY(64)'
+--let $list_sentinel = 'ABCDEF'
+--let $number_of_data_types = `SELECT FIND_IN_SET($list_sentinel, CONCAT_WS(',', $data_type_list, $list_sentinel)) - 1`
+--let $data_type_index = 1
+
+--let $server_1_datadir = `SELECT @@datadir`
+--source include/rpl_connection_slave.inc
+--let $server_2_datadir = `SELECT @@datadir`
+--source include/rpl_connection_master.inc
+
+while($data_type_index <= $number_of_data_types)
+{
+  --let $data_type = `SELECT ELT($data_type_index, $data_type_list)`
+  --echo ********************************************************************************
+  --echo Testing '$data_type' data type
+  --echo ********************************************************************************
+  --echo
+
+  --eval CREATE TABLE t1(id BIGINT, col $data_type) ENGINE=InnoDB
+
+  --source include/sync_slave_sql_with_master.inc
+  ALTER TABLE t1 ENGINE=Blackhole;
+
+  --source include/rpl_connection_master.inc
+  INSERT INTO t1 VALUES(1, 'testblob_1');
+  FLUSH BINARY LOGS;
```

Comment:
Why so many binlog flushes are required and why they are required at all?

Reply:
The idea was to have a separate, currently closed binlog.nnn file which would contain only one `Update_rows` event. This helps to add stability/predictability when we extract event data via
`--exec $MYSQL_BINLOG -vv --base64-output=decode-rows ... | grep ... > ...`

Reply:
Maybe mysqlbinlog arguments to select the desired event only (--start-position/--stop-position) would be more robust? The current way looks like it could be broken by GTID events, for example

Reply:
Removed all "FLUSH BINARY LOG". Reworked record extraction - now we use --start-position/--stop-position.